content/departments/product-engineering/engineering/code-graph/search/guiding_principles.md

```diff
@@ -0,0 +1,41 @@
+# Guiding principles for teamwork on Search Product
+
+This document outlines some guiding principles we seek to follow when welcoming engineers onto the Search Product team. It also serves as shared vocabulary on the team, when we talk about the culture we try to foster.
+
+## You're trusted by default
+
+The notion that one must "earn the trust" of their peers or manager is extremely limiting for newcomers, who are faced with this arbitrary "bar of trust" they must meet before they can deploy agency.
+
+On this team, trust is a given. Your teammates will value the opinions, questions, ideas and concerns you will bring forward from day zero. When we ask you questions, it's out of genuine interest for what you might have to answer, not as a test of your worth, and your answers will not be judged.
+
+## Liberating ability through context
+
+We trust in your skills and ability; this is why you are on this team. But raw ability must be coupled with context on the team's technical domain and its goals in order to produce impact.
+
+Our job as you onboard is to give you this context, so that you can liberate your ability. As a result, your job as you onboard is to focus on absorbing context, rather than attempting to demonstrate ability through immediate execution. This doesn't mean operating solely in the abstract, as of course part of absorbing context will entail tackling some practical problems.
```

Comment:
Yeah this was a tricky one. Some version of it seemed to make sense when we were discussing it with @camdencheek, but it proved tricky to capture in a truthful and non-hand wavy way.

I don't want to be _too_ particular about specifying precisely what "context" is, or how long it'll take to have it, because that will vary a lot from individual to individual, and the important bit is that one should bias on learning rather than executing in their first few weeks.

So, let's try again, wdyt of this?

```
As you onboard, our focus is to equip you to do your best work. For us, this means providing extra care to help you understand how the team works and  communicates, what our goals and priorities are, and any other useful information (technical bigger picture, historical decisions, etc.). During these first few weeks, your focus should be on learning, not executing. We believe that you'll be in the best position to execute if you have a good grasp of tying tasks back to this wider context.
```

Reply:
ship it!!1

Reply:
> and the important bit is that one should bias on learning rather than executing in their first few weeks.

yes